src/lib/extended/class/class.ts

```diff
@@ -12,155 +12,80 @@ import {
+      this.ngClassInstance = new NgClass(
+        this.iterableDiffers, this.keyValueDiffers, this.elementRef, this.renderer
       );
     }
+    this.marshal.init(this.nativeElement, this.DIRECTIVE_KEY, this.updateWithValue.bind(this));
+  }
+
+  protected updateWithValue(value: any) {
+    this.ngClassInstance.ngClass = value;
+    this.ngClassInstance.ngDoCheck();
   }
 
   // ******************************************************************
   // Lifecycle Hooks
   // ******************************************************************
 
-  /**
-   * For @Input changes on the current mq activation property
-   */
-  ngOnChanges(changes: SimpleChanges) {
-    if (this._base.activeKey in changes) {
-      this._ngClassInstance.ngClass = this._base.mqActivation.activatedInput || '';
-    }
-  }
-
-  ngOnInit() {
-    this._configureMQListener();
-  }
-
   /**
    * For ChangeDetectionStrategy.onPush and ngOnChanges() updates
    */
   ngDoCheck() {
-    this._ngClassInstance.ngDoCheck();
+    this.ngClassInstance.ngDoCheck();
   }
+}
 
-  ngOnDestroy() {
-    this._base.ngOnDestroy();
-  }
+const inputs = [
+  'ngClass', 'ngClass.xs', 'ngClass.sm', 'ngClass.md', 'ngClass.lg', 'ngClass.xl',
+  'ngClass.lt-sm', 'ngClass.lt-md', 'ngClass.lt-lg', 'ngClass.lt-xl',
+  'ngClass.gt-xs', 'ngClass.gt-sm', 'ngClass.gt-md', 'ngClass.gt-lg'
+];
 
-  // ******************************************************************
-  // Internal Methods
-  // ******************************************************************
+const selector = `
+  [ngClass], [ngClass.xs], [ngClass.sm], [ngClass.md], [ngClass.lg], [ngClass.xl],
+  [ngClass.lt-sm], [ngClass.lt-md], [ngClass.lt-lg], [ngClass.lt-xl],
+  [ngClass.gt-xs], [ngClass.gt-sm], [ngClass.gt-md], [ngClass.gt-lg]
+`;
 
-  /**
-   * Build an mqActivation object that bridges mql change events to onMediaQueryChange handlers
-   * NOTE: We delegate subsequent activity to the NgClass logic
-   *       Identify the activated input value and update the ngClass iterables...
-   *       Use ngDoCheck() to actually apply the values to the element
-   */
-  protected _configureMQListener(baseKey = 'ngClass') {
-    const fallbackValue = this._base.queryInput(baseKey);
-    this._base.listenForMediaQueryChanges(baseKey, fallbackValue, (changes: MediaChange) => {
-      this._ngClassInstance.ngClass = changes.value || '';
-      this._ngClassInstance.ngDoCheck();
-    });
-  }
-
-  /**
-   * Special adapter to cross-cut responsive behaviors and capture mediaQuery changes
-   * Delegate value changes to the internal `_ngClassInstance` for processing
-   */
-  protected _base: BaseDirectiveAdapter;   // used for `ngClass.xxx` selectors
+/**
+ * Directive to add responsive support for ngClass.
+ * This maintains the core functionality of 'ngClass' and adds responsive API
+ * Note: this class is a no-op when rendered on the server
+ */
+@Directive({selector, inputs})
+export class DefaultClassDirective extends ClassDirective {
```

Comment:
I assume you are doing this to preserve support for deprecated `ClassDirective` while introduce the new `DefaultClassDirective` ?
If yes, this means that after the deprecated API is deleted, the `DefaultClassDirective` is the only directive?

Reply:
`ClassDirective` (and other similar non-directives) are not deprecated, and are still being exported. They just aren't Angular `Directive`s anymore. They're used when you want to create a new custom breakpoint directive, like `DefaultClassDirective`.